src/components/forms/TextInput.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { throttle, debounce } from 'throttle-debounce';
+import IconInput from 'components/forms/IconInput';
+import DecoratedInput from 'components/forms/DecoratedInput';
+
+export const throttleValue = 500;
+
+const TextInput = ({
```

Comment:
Needs unit tests

Reply:
Hmmm so I can do snapshot tests as an easy fix. I could "also" test the throttling functions but I'm hesitant to test that, unless we're confident that we don't want to change the throttling configuration.

Reply:
Snapshots to cover different prop combinations would work. For throttling, maybe spy on `onThrottledChange` - Jest timers may be useful here. If it's too much of a pain, snapshots might be enough.

Reply:
I'm now testing that onThrottldeChange is now firing. I'm opting to test the function I'm using directly to test whether the debounce vs throttle variant of the function fires.